tests/python/test_tools_smoke.py

```diff
@@ -0,0 +1,371 @@
+
+    def test_hardirqs(self):
+        self.run_with_duration("hardirqs.py 1 1")
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_killsnoop(self):
+        # Because killsnoop intercepts signals, if we send it a SIGINT we we
+        # we likely catch it while it is handling the data packet from the
+        # BPF program, and the exception from the SIGINT will be swallowed by
+        # ctypes. Therefore, we use SIGKILL.
+        # To reproduce the above issue, run killsnoop and in another shell run
+        # `kill -s SIGINT $(pidof python)`. As a result, killsnoop will print
+        # a traceback but will not exit.
+        self.run_with_int("killsnoop.py", kill=True)
+
+    @skipUnless(kernel_version_ge(4,9), "requires kernel >= 4.9")
+    def test_llcstat(self):
+        # Requires PMU, which is not available in virtual machines.
+        pass
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_mdflush(self):
+        self.run_with_int("mdflush.py")
+
+    @skipUnless(kernel_version_ge(4,6), "requires kernel >= 4.6")
+    def test_memleak(self):
+        self.run_with_duration("memleak.py 1 1")
+
+    @skipUnless(kernel_version_ge(4,8), "requires kernel >= 4.8")
+    def test_mountsnoop(self):
+        self.run_with_int("mountsnoop.py")
+
+    @skipUnless(kernel_version_ge(4,3), "requires kernel >= 4.3")
+    def test_mysqld_qslower(self):
+        # Deliberately left empty -- mysqld_qslower requires an instance of
+        # MySQL to be running, or it fails to attach.
+        pass
+
+    @skipUnless(kernel_version_ge(4,6), "requires kernel >= 4.6")
+    def test_offcputime(self):
+        self.run_with_duration("offcputime.py 1")
+
+    @skipUnless(kernel_version_ge(4,6), "requires kernel >= 4.6")
+    def test_offwaketime(self):
+        self.run_with_duration("offwaketime.py 1")
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_oomkill(self):
+        self.run_with_int("oomkill.py")
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_opensnoop(self):
+        self.run_with_int("opensnoop.py")
+
+    def test_pidpersec(self):
+        self.run_with_int("pidpersec.py")
+
+    @skipUnless(kernel_version_ge(4,9), "requires kernel >= 4.9")
+    def test_profile(self):
+        self.run_with_duration("profile.py 1")
+
+    def test_runqlat(self):
+        self.run_with_duration("runqlat.py 1 1")
+
+    @skipUnless(kernel_version_ge(4,9), "requires kernel >= 4.9")
+    def test_runqlen(self):
+        self.run_with_duration("runqlen.py 1 1")
+
+    def test_slabratetop(self):
+        self.run_with_duration("slabratetop.py 1 1")
+
+    def test_softirqs(self):
+        # TODO Temporary disabled as softirqs.py doesn't work on recent
+        # kernels (can't find some of its attach targets). Need to revisit
+        # it to use the softirq tracepoints. Tracked in bcc#1031.
```

Comment:
Let's start with one file, we can always break it up later if it becomes annoying.